lib/tasks/mfa_notification.rake

```diff
@@ -0,0 +1,27 @@
+require "resolv"
+
+def mx_exists?(email)
+  domain = email.split("@").last
+  mx_resolver = Resolv::DNS.new
+  mx_resolver.timeouts = 10
+
+  return false if mx_resolver.getresources(domain, Resolv::DNS::Resource::IN::MX).empty?
+  true
+rescue StandardError => e
+  puts "Error during processing: #{$ERROR_INFO}"
+  puts "Backtrace:\n\t#{e.backtrace.join("\n\t")}"
+  false
+end
+
+namespace :mfa_notification do
+  desc "Send email notification to all users about MFA"
+  task send: :environment do
+    i = 0
+    total = User.count
+    User.find_each do |user|
```

Comment:
Consider:

- skipping users who already have MFA enabled for both UI+API.
- limit to sending in small batches of users over a few days/weeks, allowing time for feedback on the email content from the early recipients.

Reply:
Thanks. We will send mail to users with login in 2019 first and possible after two weeks for rest.